src/main/java/org/truffleruby/core/fiber/FiberManager.java

```diff
@@ -135,18 +136,20 @@ private void fiberMain(RubyContext context, DynamicObject fiber, DynamicObject b
 
         } catch (KillException e) {
             // Naturally exit the Fiber on catching this
+        } catch (ExitException e) {
+            sendExceptionToParentFiber(fiber, e);
+        } catch (RaiseException e) {
```

Comment:
I don't really care that these are separated, but you could do a multi-catch, since they have the same body.

Reply:
It's to keep it in the same order as in the ThreadManager, where there the handlers differ.